nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/nucleusdata/Kit.java

```diff
@@ -135,6 +135,26 @@ default Kit addCommand(String command) {
     Kit updateKitInventory(Player player);
 
     /**
+     * Redeems any items and commands in this kit for the specified player.
+     *
+     * @param player The player.
+     * @return <code>true</code> if the redemption was successful
+     */
+    default boolean redeem(Player player) {
+        boolean success = redeemKitItems(player);
+        if (success) redeemKitCommands(player);
+        return success;
```

Comment:
No. While there might be merit with adding an enum or result object that returns what was actually done, when returning a boolean, this is correct as it tells you whether the configured behaviour executed, which might include the fact some items weren't redeemed.